src/ui/public/vislib/lib/axis.js

```diff
@@ -1,695 +1,227 @@
 import d3 from 'd3';
-import $ from 'jquery';
 import _ from 'lodash';
-import moment from 'moment';
-import VislibLibErrorHandlerProvider from 'ui/vislib/lib/_error_handler';
-import errors from 'ui/errors';
+import $ from 'jquery';
+import ErrorHandlerProvider from 'ui/vislib/lib/_error_handler';
+import AxisTitleProvider from 'ui/vislib/lib/axis_title';
+import AxisLabelsProvider from 'ui/vislib/lib/axis_labels';
+import AxisScaleProvider from 'ui/vislib/lib/axis_scale';
 
 export default function AxisFactory(Private) {
-
-  const ErrorHandler = Private(VislibLibErrorHandlerProvider);
-
+  const ErrorHandler = Private(ErrorHandlerProvider);
+  const AxisTitle = Private(AxisTitleProvider);
+  const AxisLabels = Private(AxisLabelsProvider);
+  const AxisScale = Private(AxisScaleProvider);
+  const defaults = {
+    show: true,
+    type: 'value',
+    elSelector: '.axis-wrapper-{pos} .axis-div',
+    position: 'left',
+    axisFormatter: null, // TODO: create default axis formatter
+    scale: 'linear',
```

Comment:
Maybe we can call this `scaleType`? Since `this.scale` is actually an instance of `AxisScale`